### ISSUE TITLE
Enable TLS in tonic

### ISSUE THREAD

**Issue by picklenerd** [boltlabs-inc/key-mgmt#198]
TLS isn't currently enabled in `tonic`. There are lots of examples for how to set it up in the `tonic` repo and docs.

https://docs.rs/tonic/latest/tonic/transport/index.html
https://github.com/hyperium/tonic/blob/master/examples/src/tls/client_rustls.rs
https://github.com/hyperium/tonic/blob/master/examples/src/tls/server_rustls.rs

**Comment by picklenerd** [comment]
Went on a long adventure with those tonic examples above. We might need to do that down the line if we want to customize TLS more, but these examples still use rustls and should be fine.

https://github.com/hyperium/tonic/blob/master/examples/src/tls/client.rs
https://github.com/hyperium/tonic/blob/master/examples/src/tls/server.rs

**Comment by picklenerd** [comment]
The adventure continues. The simple case doesn't seem to play nice with self-signed certs so I experimented with making a local CA and signing the certs with that. I still couldn't get it to work. Eventually I pulled in the code from the `transport` crate and made some progress, but I'm still tracking down TLS errors.

The current error is coming from `tonic` after we call its generated functions. I get `client requires absolute-form URIs`. All of the URIs I look at have the right form so I need to figure out where this error is coming from. It's not in the generated code so it must be in `tonic` somewhere.
